src/app/loadout-drawer/loadout-apply.ts

```diff
@@ -1220,16 +1222,22 @@ function applyLoadoutMods(
       const pluggingSteps = createPluggingStrategy(defs, item, assignments);
       const assignmentSequence = pluggingSteps.filter((assignment) => assignment.required);
       infoLog(TAG, 'Applying mods', assignmentSequence, 'to', item.name);
-      if (assignmentSequence) {
-        applyModsPromises.push(
-          dispatch(
-            equipModsToItem(item, assignmentSequence, handleSuccess, handleFailure, cancelToken),
-          ),
-        );
+
+      if (assignmentSequence.length) {
+        const [f, m] = _.partition(assignmentSequence, (a) => isFashionPlug(a.mod));
+        modAssigns.push({ item, actions: m });
+        fashionAssigns.push({ item, actions: f });
       }
     }
 
-    await Promise.all(applyModsPromises);
+    // given limited time, slow API, impatient teammates, we'll plug all mods into all armor pieces, *then* all fashion
+    for (const assignGroup of [modAssigns, fashionAssigns]) {
+      await Promise.all(
+        assignGroup.map(({ item, actions }) =>
+          dispatch(equipModsToItem(item, actions, handleSuccess, handleFailure, cancelToken)),
+        ),
+      );
```

Comment:
I think it makes sense to *sort* fashion behind mods, but it might not be the best to do them as two entirely distinct sections - this makes a single slow mod application block all fashion. Something that might matter more as we increase parallelism.

Reply:
right now a slow mod already blocks all fashion. only one mod can be applying at once, period.

this can and will be changed when application is parallel but right now it's not.